core/src/com/unciv/logic/battle/BattleDamage.kt

```diff
@@ -70,15 +67,51 @@ object BattleDamage {
                 if (civResources[resource]!! < 0 && !civInfo.isBarbarian())
                     modifiers["Missing resource"] = -25
 
-
-            val nearbyCivUnits = combatant.unit.getTile().getTilesInDistance(2)
-                .flatMap { it.getUnits() }.filter { it.civInfo == combatant.unit.civInfo }
-            if (nearbyCivUnits.any { it.hasUnique("Bonus for units in 2 tile radius 15%") }) {
-                val greatGeneralModifier =
-                    if (combatant.unit.civInfo.hasUnique("Great General provides double combat bonus")) 30 else 15
-                modifiers["Great General"] = greatGeneralModifier
+            // Deprecated since 3.15.5
+                val nearbyCivUnits = combatant.unit.getTile().getTilesInDistance(2)
+                    .flatMap { it.getUnits() }.filter { it.civInfo == combatant.unit.civInfo }
+                if (nearbyCivUnits.any { it.hasUnique("Bonus for units in 2 tile radius 15%") }) {
+                    val greatGeneralModifier =
+                        if (combatant.unit.civInfo.hasUnique("Great General provides double combat bonus")) 30 else 15
+                    modifiers["Great General"] = greatGeneralModifier
+                }
+            //
+            
+            // Calculate strength bonus from nearby great generals/great admirals or mod equivalents
+            val nearbyBonusUnits = combatant.unit.getTile().getTilesInDistance(combatant.getCivInfo().gameInfo.ruleSet.modOptions.maxGreatGeneralSearchDistance)
+                .flatMap { it.getUnits() }
+                .filter { it.civInfo == combatant.unit.civInfo && it.hasUnique("+[]% Strength for [] units within [] tiles")}
+            
+            for (unit in nearbyBonusUnits) {
+                var totalApplicableBonus = 0
+                for (unique in unit.getMatchingUniques("+[]% Strength for [] units within [] tiles")) 
+                    if (combatant.unit.matchesFilter(unique.params[1]) && HexMath.getDistance(combatant.unit.getTile().position, unit.getTile().position) <= unique.params[2].toInt())
```

Comment:
combatant.unit.getTile().aerialDistanceTo(unit.getTile())
HexMath.getDistance should be deprecated since it doesn't consider world-wrapped coordinates